test/python/test_shallow_water_state.py

```diff
@@ -0,0 +1,171 @@
+        assert_array_equal(state.u.data[xps-xms:xpe-xms+1, yms-yms], 10.0 * south)
+        assert_array_equal(state.v.data[xps-xms:xpe-xms+1, yms-yms], 20.0 * south)
+        assert_array_equal(state.h.data[xps-xms:xpe-xms+1, yms-yms], 30.0 * south)
+    if (west != -1):
+        assert_array_equal(state.u.data[xms-xms, yps-yms:ype-yms+1],  10.0 * west)
+        assert_array_equal(state.v.data[xms-xms, yps-yms:ype-yms+1],  20.0 * west)
+        assert_array_equal(state.h.data[xms-xms, yps-yms:ype-yms+1],  30.0 * west)
+    if (east != -1):
+        assert_array_equal(state.u.data[xme-xms, yps-yms:ype-yms+1], 10.0 * east)
+        assert_array_equal(state.v.data[xme-xms, yps-yms:ype-yms+1], 20.0 * east)
+        assert_array_equal(state.h.data[xme-xms, yps-yms:ype-yms+1], 30.0 * east)
+
+@pytest.mark.mpi(min_size=1)
+def test_shallow_water_state_read(input_io_geometry, input_gt4py_config):
+    npx = input_io_geometry.npx
+    npy = input_io_geometry.npy
+    xps = input_io_geometry.xps
+    xpe = input_io_geometry.xpe
+    yps = input_io_geometry.yps
+    ype = input_io_geometry.ype
+    xms = input_io_geometry.xms
+    xme = input_io_geometry.xme
+    yms = input_io_geometry.yms
+    yme = input_io_geometry.yme
+    clock = 182850.615187004
+    u = np.full((npx, npy), 5.0)
+    v = np.full((npx, npy), 6.0)
+    h = np.full((npx, npy), 7.0)
+
+    state = ShallowWaterState(input_io_geometry, input_gt4py_config)
+    state.read("../test_input/test_shallow_water_reader.nc")
+
+    assert_almost_equal(state.clock, clock, decimal=9)
+    assert_array_equal(state.u.data[xps-xms:xpe-xms+1, yps-yms:ype-yms+1], u)
+    assert_array_equal(state.v.data[xps-xms:xpe-xms+1, yps-yms:ype-yms+1], v)
+    assert_array_equal(state.h.data[xps-xms:xpe-xms+1, yps-yms:ype-yms+1], h)
+
+@pytest.mark.mpi(min_size=1)
+def test_shallow_water_state_write(input_io_geometry, input_gt4py_config):
+    npx = input_io_geometry.npx
+    npy = input_io_geometry.npy
+    xps = input_io_geometry.xps
+    xpe = input_io_geometry.xpe
+    yps = input_io_geometry.yps
+    ype = input_io_geometry.ype
+    xms = input_io_geometry.xms
+    xme = input_io_geometry.xme
+    yms = input_io_geometry.yms
+    yme = input_io_geometry.yme
+    clock = 1000.0
+    u = np.full((npx, npy), 5.0)
+    v = np.full((npx, npy), 6.0)
+    h = np.full((npx, npy), 7.0)
+
+    state = ShallowWaterState(input_io_geometry, input_gt4py_config, u=u, v=v, h=h, clock=clock)
+    state.write("../test_output/test_shallow_water_writer.nc")
+
+    state_read = ShallowWaterState(input_io_geometry, input_gt4py_config)
+    state_read.read("../test_output/test_shallow_water_writer.nc")
+
+    assert_almost_equal(state.clock, state_read.clock, decimal=9)
+
+    assert_array_equal(state_read.u.data[xps-xms:xpe-xms+1, yps-yms:ype-yms+1], u)
+    assert_array_equal(state_read.v.data[xps-xms:xpe-xms+1, yps-yms:ype-yms+1], v)
+    assert_array_equal(state_read.h.data[xps-xms:xpe-xms+1, yps-yms:ype-yms+1], h)
```

Comment:
Need a newline here

Reply:
@christopherwharrop-noaa I just caught that as well.